### DISCUSSION TITLE
Milvus inserts a vector. Why is there a binary in the data structure? Is it possible to store the information you want?

### DISCUSSION THREAD

**Discussion by computerAItest** [milvus-io/milvus#4069] [Q&A and General discussion] [answered]
Milvus inserts a vector. Why is there a binary in the data structure? Is it possible to store the information you want?

type Entity struct {
	FloatData  []float32
	BinaryData []byte
}

**Comment by sahuang** [comment, marked as answer]
Currently, Milvus supports two types of vectors: float vectors and binary vectors. If you wish to insert float vectors, use `FloatData []float32`. Otherwise use `BinaryData []byte` for binary vectors.
